crates/types/src/partition_table.rs

```diff
@@ -94,8 +94,8 @@ impl PartitionTable {
         self.partitions.iter_mut()
     }
 
-    pub fn num_partitions(&self) -> u64 {
-        u64::try_from(self.partitions.len()).expect("number of partitions should fit into u64")
+    pub fn num_partitions(&self) -> u16 {
+        u16::try_from(self.partitions.len()).expect("number of partitions should fit into u16")
```

Comment:
Should we add a check to the `PartitionTableBuilder` which prevents one from creating a `PartitionTable` with more than `u16` partitions?

Reply:
Good call, will add.